lib/gringotts/gateways/securion_pay.ex

```diff
@@ -0,0 +1,217 @@
+  | `customerId`             | Unique identifier of the customer                                                             |
+
+  ## Registering your SecurionPay account at `Gringotts`
+
+  After [making an account on SecurionPay][SP], find your `Secret key` at [Account Settings][api-key] 
+
+  Your Application config **must include the `:secret_key`  field**.
+  It would look something like this:
+
+      config :gringotts, Gringotts.Gateways.SecurionPay,
+        secret_key: "your_secret_key"
+
+  [SP]: https://securionpay.com/
+  [api-key]: https://securionpay.com/account-settings#api-keys
+
+  ## Supported countries
+
+  SecurionPay supports the countries listed [here][country-list]
+
+  [country-list]: https://securionpay.com/supported-countries-businesses/
+  """
+
+  @base_url "https://api.securionpay.com/"
+  # The Base module has the (abstract) public API, and some utility
+  # implementations.
+
+  use Gringotts.Gateways.Base
+
+  # The Adapter module provides the `validate_config/1`
+  # Add the keys that must be present in the Application config in the
+  # `required_config` list
+  use Gringotts.Adapter, required_config: [:secret_key]
+
+  import Poison, only: [decode: 1]
+
+  alias Gringotts.{CreditCard, Response, Address}
+
+  @doc """
+  Authorizes a credit card transaction.
+
+  The authorization validates the `card` details with the banking network,
+  places a hold on the transaction `amount` in the customer’s issuing bank and
+  also triggers risk management. Funds are not transferred.
+
+  The second argument can be a CreditCard or a cardId. The customerId of the cutomer who owns the card must be
+  given in optional field. 
+
+  To transfer the funds to merchant's account follow this up with a `capture/3`.
+
+  SecurionPay returns a `chargeId` which uniquely identifies a transaction (available in the `Response.id` field) 
+  which should be stored at your side to use in:
+
+  * `capture/3` an authorized transaction.
+  * `void/2` a transaction.
+
+  ## Example 1
+      iex> amount = Money.new(20, :USD}
+      iex> opts = [config: [:secret_key: "c2tfdGVzdF9GZjJKcHE1OXNTV1Q3cW1JOWF0aWk1elI6"]]
+      iex> card = %CreditCard{
+           first_name: "Harry",
+           last_name: "Potter",
+           number: "4200000000000000",
+           year: 2027,
+           month: 12,
+           verification_code: "123",
+           brand: "VISA"
+          }
+      iex> result = Gringotts.Gateways.SecurionPay.authorize(amount, card, opts)
+
+  ## Example 2
+      iex> amount = Money.new(20, :USD}
+      iex> opts = [config: [:secret_key: "c2tfdGVzdF9GZjJKcHE1OXNTV1Q3cW1JOWF0aWk1elI6"], customerId: "cust_zpYEBK396q3rvIBZYc3PIDwT"]
+      iex> card = "card_LqTT5tC10BQzDbwWJhFWXDoP"
+      iex> result = Gringotts.Gateways.SecurionPay.authorize(amount, card, opts)
+
+  """
+  @spec authorize(Money.t(), CreditCard.t() | {}, keyword) :: {:ok | :error, Response}
```

Comment:
IMO, the `authorize` spec should be, following the work in `Stripe` and `WireCard`,
```elixir
@spec authorize(Money.t(), CreditCard.t() | String.t(), keyword) :: {:ok | :error, Response.t()}
```
That is, the `card` argument can be either a `CreditCard` struct or a string (`card_token`). Other tokens, if needed, can be passed via `opts`.